### PR TITLE
[3.0] Make HAProxy work as an http proxy instead of a tcp proxy.

### PR DESCRIPTION
**Pull request by ereslibre**
This allows us to add fine-grained timeouts depending on the endpoint being
accessed or with what parameters (e.g. /log?follow=true should have no
timeout as happens on the apiserver). /exec is another example, but in this
case the protocol is upgraded to spdy.

Fixes: bsc#1071994
(cherry picked from commit 442a76cad214f2308f6a1de0ddef8febca8074c8)

### CODE REVIEW

**Comment by ereslibre** [comment]
[Bug 1071994 - kubectl run --attach exits early](https://trello.com/c/nNNdxJAn/704-bug-1071994-kubectl-run-attach-exits-early)

**Comment by ereslibre** [comment]
Please, do not merge: figuring out why the two PR's together aren't passing in Jenkins: https://github.com/kubic-project/jenkins-library/pull/170

File: salt/_macros/kubectl.jinja [hunk 1]
```diff
@@ -91,10 +91,10 @@
 wait-for-{{ deployment }}-deployment:
```
Comment (inercia) [inline salt/_macros/kubectl.jinja]: Not for this PR, but at some moment we should refactor this and create a `wait_for_deployment` state in Python.

**Comment by ereslibre** [comment]
@flavio @kiall: We need to force merge this one. Passing on https://github.com/kubic-project/jenkins-library/pull/174

**Pull request merged**